fdbsyncd/fdbsync.cpp

```diff
@@ -0,0 +1,603 @@
+        return;
+
+	SWSS_LOG_NOTICE("%sIMET Del route key:%s vtep:%s %s", 
+			m_AppRestartAssist->isWarmStartInProgress() ? "WARM-RESTART:" : "", 
+			key.c_str(), inet_ntoa(vtep), vlan_id.c_str());
+
+    std::vector<FieldValueTuple> fvVector;
+    FieldValueTuple f("vni", to_string(vni));
+    fvVector.push_back(f);
+
+    // If warmstart is in progress, we take all netlink changes into the cache map
+    if (m_AppRestartAssist->isWarmStartInProgress())
+    {
+        m_AppRestartAssist->insertToMap(APP_VXLAN_REMOTE_VNI_TABLE_NAME, key, fvVector, true);
+        return;
+    }
+    
+	m_imetTable.del(key);
+    return;
+}
+
+void FdbSync::macUpdateVxlanDB(string key)
+{
+    string vtep = m_mac[key].vtep;
+    string type;
+    string vni = to_string(m_mac[key].vni);
+
+    if (m_mac[key].type)
+        type = "static";
+    else
+        type = "dynamic";
+
+    std::vector<FieldValueTuple> fvVector;
+    FieldValueTuple rv("remote_vtep", vtep);
+    FieldValueTuple t("type", type);
+    FieldValueTuple v("vni", vni);
+    fvVector.push_back(rv);
+    fvVector.push_back(t);
+    fvVector.push_back(v);
+
+	SWSS_LOG_NOTICE("%sVXLAN_FDB_TABLE: ADD_KEY %s vtep:%s type:%s", 
+			m_AppRestartAssist->isWarmStartInProgress() ? "WARM-RESTART:" : "" ,
+			key.c_str(), vtep.c_str(), type.c_str());
+
+    // If warmstart is in progress, we take all netlink changes into the cache map
+    if (m_AppRestartAssist->isWarmStartInProgress())
+    {
+        m_AppRestartAssist->insertToMap(APP_VXLAN_FDB_TABLE_NAME, key, fvVector, false);
+        return;
+    }
+    
+	SWSS_LOG_NOTICE("VXLAN_FDB_TABLE: ADD_KEY %s vtep:%s type:%s", key.c_str(), vtep.c_str(), type.c_str());
+    m_fdbTable.set(key, fvVector);
+}
+
+void FdbSync::macDelVxlanDB(string key)
+{
+    string vtep = m_mac[key].vtep;
+    string type;
+    string vni = to_string(m_mac[key].vni);
+
+    if (m_mac[key].type)
+        type = "static";
+    else
+        type = "dynamic";
+
+    std::vector<FieldValueTuple> fvVector;
+    FieldValueTuple rv("remote_vtep", vtep);
+    FieldValueTuple t("type", type);
+    FieldValueTuple v("vni", vni);
+    fvVector.push_back(rv);
+    fvVector.push_back(t);
+    fvVector.push_back(v);
+
+	SWSS_LOG_NOTICE("%sVXLAN_FDB_TABLE: DEL_KEY %s vtep:%s type:%s", 
+			m_AppRestartAssist->isWarmStartInProgress() ? "WARM-RESTART:" : "" ,            
+			key.c_str(), vtep.c_str(), type.c_str());
+
+    // If warmstart is in progress, we take all netlink changes into the cache map
+    if (m_AppRestartAssist->isWarmStartInProgress())
+    {
+        m_AppRestartAssist->insertToMap(APP_VXLAN_FDB_TABLE_NAME, key, fvVector, true);
+        return;
+    }
+    
+	SWSS_LOG_NOTICE("VXLAN_FDB_TABLE: DEL_KEY %s vtep:%s type:%s", key.c_str(), vtep.c_str(), type.c_str());
+    m_fdbTable.del(key);
+    return;
+
+}
+
+void FdbSync::macAddVxlan(string key, struct in_addr vtep, uint32_t type, uint32_t vni, string intf_name)
+{
+    m_mac[key].vtep   =  inet_ntoa(vtep);

```

Comment:
You could optimize it by `m_mac[key] = { vtep, type, vni, intf_name }` and use the same function to write to DB so as to avoid number of functions.